### PR TITLE
Fixes #211

### LINKED ISSUE #211: logger option genReqId not working after lifecycle refactor.

### PR DESCRIPTION
**Pull request by delvedor**
As titled.

### CODE REVIEW

File: test/logger.test.js [hunk 1]
```diff
@@ -124,3 +124,32 @@ test('expose the logger', t => {
...
+    logger: {
+      genReqId: function () {
+        return 'a'
+      }
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    t.ok(req.req.id)
+    reply.send({ id: req.req.id })
```
Comment (mcollina) [inline test/logger.test.js]: IMHO we should expose this in our wrapped request.

File: fastify.js [hunk 2]
```diff
@@ -40,7 +40,8 @@ function build (options) {
   const map = new Map()
 
   // logger utils
-  const genReqId = loggerUtils.reqIdGenFactory()
+  const customGenReqId = options.logger ? options.logger.genReqId : null
+  const genReqId = loggerUtils.reqIdGenFactory(customGenReqId)
```
Comment (allevo) [inline fastify.js]: IMHO passing the function to return it is useless
```js
const genReqId = (options.logger ? options.logger.genReqId : null) || loggerUtils.reqIdGenFactory()
```
So the `loggerUtils.reqIdGenFactory` is without params

File: test/logger.test.js (hunk 1 above)
Comment (delvedor) [reply test/logger.test.js]: Why?

Comment (mcollina) [reply test/logger.test.js]: It seems an interesting info. Maybe not.

Comment (delvedor) [reply test/logger.test.js]: We can discuss this once we add also the `header` object to request, it is more *in scope* discussion :)

Comment (mcollina) [reply test/logger.test.js]: ok

Comment (mcollina) [reply test/logger.test.js]: +1

**Pull request merged**